teku/src/test/java/tech/pegasys/teku/cli/options/ReflectionBasedBeaconRestApiOptionsTest.java

```diff
@@ -65,6 +65,66 @@ public void restApiEnabled_shouldNotRequireAValue() {
         .isEqualTo(tekuConfiguration);
   }
 
+  @Test
+  public void restApiEnabledAndPortNotSpecified_shouldProvideDefaults() {
+    TekuConfiguration tekuConfiguration = getTekuConfigurationFromArguments();
+    final BeaconRestApiConfig config = getConfig(tekuConfiguration);
+    assertThat(config.isRestApiEnabled()).isFalse();
+    assertThat(config.getRestApiPort()).isEqualTo(5051);
+    assertThat(
+            createConfigBuilder().restApi(b -> b.restApiEnabled(false).restApiPort(5051)).build())
+        .usingRecursiveComparison()
+        .isEqualTo(tekuConfiguration);
+  }
+
+  @Test
+  public void restApiEnabledNullAndPortSpecified_shouldEnable() {
+    TekuConfiguration tekuConfiguration = getTekuConfigurationFromArguments("--rest-api-port=5051");
+    final BeaconRestApiConfig config = getConfig(tekuConfiguration);
+    assertThat(config.isRestApiEnabled()).isTrue();
+    assertThat(config.getRestApiPort()).isEqualTo(5051);
+    assertThat(createConfigBuilder().restApi(b -> b.restApiEnabled(true).restApiPort(5051)).build())
+        .usingRecursiveComparison()
+        .isEqualTo(tekuConfiguration);
+  }
+
+  @Test
+  public void restApiEnabledAndPortNotSpecified_shouldProvideDefault() {
+    TekuConfiguration tekuConfiguration =
+        getTekuConfigurationFromArguments("--rest-api-enabled=true");
+    final BeaconRestApiConfig config = getConfig(tekuConfiguration);
+    assertThat(config.isRestApiEnabled()).isTrue();
+    assertThat(config.getRestApiPort()).isEqualTo(5051);
+    assertThat(createConfigBuilder().restApi(b -> b.restApiEnabled(true).restApiPort(5051)).build())
+        .usingRecursiveComparison()
+        .isEqualTo(tekuConfiguration);
+  }
+
+  @Test
+  public void restApiEnabledAndPortSpecified_shouldHaveSpecifiedValues() {
+    TekuConfiguration tekuConfiguration =
+        getTekuConfigurationFromArguments("--rest-api-enabled=true", "--rest-api-port=5051");
```

Comment:
make this non default so we can distinguish between loading the default and loading the non default